web/yo/app/scripts/controllers/reportgenerator.js

```diff
@@ -204,4 +204,264 @@ angular.module('oncokbApp')
+                                drugs: drugs.join(), 
+                                level: item.levelOfEvidence,
+                                description: item.description,
+                                citations: getCitations(item.description)
+                            });
+                        }
+                    } 
+                    loopArray(arr, ++index);
+                } else {
+                    $scope.reviewedData.drugs = reviewedData.drugs;
+                    finishLoadingReviewedData();
+                }
+            }
+            function finishLoadingReviewedData() {
+                $scope.loadingReviewed = false;
+                $scope.displayReviewedData = true;
+            }
+            $scope.updateReviewData = function() {
+                $scope.displayReviewedData = false;
+            }
+            $scope.generateEvidences = function () {
+                $scope.loadingReviewed = true;
+                reviewedData = {
+                    gene_type: [],
+                    mutation_effect: [],
+                    tumor_summary: [],
+                    drugs: []
+                };
+                $scope.reviewedData = {
+                    gene_type: [],
+                    mutation_effect: [],
+                    tumor_summary: [],
+                    drugs: []
+                };
+                if ($scope.evidenceType === 'gene_type') {
+                    $http.get(OncoKB.config.publicApiLink + 'genes').then(function(response) {
+                        _.each(response.data, function(item) {
+                            reviewedData.gene_type.push({
+                                gene: item.hugoSymbol,
+                                oncogene: item.oncogene,
+                                tsg: item.tsg
+                            });
+                        });
+                        $scope.reviewedData.gene_type = reviewedData.gene_type;
+                        finishLoadingReviewedData();
+                    });
+                } else if ($scope.evidenceType === 'mutation_effect') {
+                    $http.get(OncoKB.config.publicApiLink + 'evidences/lookup?source=oncotree&evidenceTypes=MUTATION_EFFECT%2C%20ONCOGENIC').then(function (response) {
+                        _.each(response.data, function (item) {
+                            var flag = false;
+                            for (var i = 0; i < reviewedData.mutation_effect.length; i++) {
+                                var evidence = reviewedData.mutation_effect[i];
+                                if (item.gene.hugoSymbol === evidence.gene && getAlterations(item.alterations) === evidence.mutation) {
+                                    flag = true;
+                                    insertEvidence(item, evidence);
+                                    break;
+                                }
+                            }
+                            if (flag === false) {
```

Comment:
1. You probably want to rename the function `insertEvidence `. The meaning is ambiguous.
2. Could you explain the logic? You updated the evidence above, but we insert the same evidence to `reviewedData.mutation_effect`

Reply:
Because for the API, Mutation Effect and Oncogenic are two different types of evidence. But for the interface, both of these two pieces of evidence belong to Mutation Effect.